src/views/dms/chat.tsx

```diff
@@ -29,6 +29,40 @@ import useRouterMarker from "../../hooks/use-router-marker";
 import { BackIconButton } from "../../components/router/back-button";
 import decryptionCacheService from "../../services/decryption-cache";
 
+// Function to handle incoming messages
+const handleIncomingMessage = async (message: NostrEvent) => {
+  if (message.kind === kinds.EncryptedDirectMessage) {
+    const userConfirmed = await promptUserForDecryption(message);
+    if (userConfirmed) {
+      const decryptedMessage = await decryptMessage(message);
+      displayMessage(decryptedMessage);
+    } else {
+      displayMessage("Decryption denied by user.");
+    }
+  } else {
+    displayMessage(message.content);
+  }
+};
+
+// Function to prompt user for decryption
+const promptUserForDecryption = (message: NostrEvent): Promise<boolean> => {
+  return new Promise((resolve) => {
+    const userResponse = window.confirm(`Do you want to decrypt this message: ${message.content}?`);
+    resolve(userResponse);
```

Comment:
The way this is now, I get bombarded with at least 50 alert windows to confirm something when opening a chat with less than 50 messages.

![image](https://github.com/user-attachments/assets/c79a251e-e119-4019-a7bf-d1f717666265)

I suspect @hzrd149 sees a problem with nip17 integration in that prior to decryption, it is not clear who's message we are decrypting as the sender is anonymous at that point.

But as there is absolutely no way of knowing not even the timestamp of the message as per nip17 that is also supposed to be fuzzy, the user will get prompted to decrypt all messages from friends or foes alike? That's not practical.

For what it's worth, I forbade the popup to appear again and clicked on decrypt-all above, which gave me no nip17 messages although there should be some on that friend's chat. I had not analyzed the code to that detail on whither these nip17 messages should be intermixed with nip4 but I suppose that's how it should work eventually.